atom/browser/atom_download_manager_delegate.cc

```diff
@@ -127,63 +127,57 @@ void AtomDownloadManagerDelegate::OnDownloadPathGenerated(
         base::BindOnce(&AtomDownloadManagerDelegate::OnDownloadSaveDialogDone,
                        base::Unretained(this), download_id, callback);
 
-    file_dialog::ShowSaveDialog(settings, std::move(dialog_promise));
     ignore_result(dialog_promise.Then(std::move(dialog_callback)));
+    file_dialog::ShowSaveDialog(settings, std::move(dialog_promise));
   } else {
     callback.Run(path, download::DownloadItem::TARGET_DISPOSITION_PROMPT,
                  download::DOWNLOAD_DANGER_TYPE_NOT_DANGEROUS, path,
                  download::DOWNLOAD_INTERRUPT_REASON_NONE);
   }
 }
 
-#if defined(MAS_BUILD)
-void AtomDownloadManagerDelegate::OnDownloadSaveDialogDone(
-    uint32_t download_id,
-    const content::DownloadTargetCallback& download_callback,
-    bool result,
-    const base::FilePath& path,
-    const std::string& bookmark)
-#else
 void AtomDownloadManagerDelegate::OnDownloadSaveDialogDone(
     uint32_t download_id,
     const content::DownloadTargetCallback& download_callback,
-    bool result,
-    const base::FilePath& path)
-#endif
```

Comment:
:+1: on dropping the `#ifdef`s

Reply:
what do we lose by dropping the `bookmark`s?

Reply:
Nothing, the `bookmark` is exposed to the user facing API.  Not the internal chromium usage